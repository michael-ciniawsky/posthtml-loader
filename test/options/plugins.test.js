'use strict'

const webpack = require('../helpers/compiler')

describe('Options', () => {
  describe('plugins', () => {
    test('{Array}', () => {
      const config = {
        loader: {
          test: /\.html$/,
          options: {
            ident: 'posthtml',
            plugins: [
              require('../fixtures/plugin')()
            ]
          }
        }
      }

      return webpack('fixture.js', config)
        .then((stats) => {
          const [module] = stats.toJson().modules

          expect(module.source).toMatchSnapshot()
        })
        .catch((err) => err)
    })

    test('{Function} - {Array}', () => {
      const config = {
        loader: {
          test: /\.html$/,
          options: {
            ident: 'posthtml',
            plugins () {
              return [
                require('../fixtures/plugin')()
              ]
            }
          }
        }
      }

      return webpack('fixture.js', config)
        .then((stats) => {
          const [module] = stats.toJson().modules

          expect(module.source).toMatchSnapshot()
        })
        .catch((err) => err)
    })

    test('{Function} - {Object}', () => {
      const config = {
        loader: {
          test: /\.html$/,
          options: {
            ident: 'posthtml',
            plugins () {
              return require('../fixtures/plugin')()
            }
          }
        }
      }

      return webpack('fixture.js', config)
        .then((stats) => {
          const [module] = stats.toJson().modules

          expect(module.source).toMatchSnapshot()
        })
        .catch((err) => err)
    })
  })
})
