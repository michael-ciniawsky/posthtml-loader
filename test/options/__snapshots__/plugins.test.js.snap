// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Options plugins {Array} 1`] = `
"export default \`<section>Hello</section>
\`"
`;

exports[`Options plugins {Function} - {Array} 1`] = `
"export default \`<section>Hello</section>
\`"
`;

exports[`Options plugins {Function} - {Object} 1`] = `
"export default \`<section>Hello</section>
\`"
`;
